File: Tutorial3-1Scope1/src/main/java/com/example/tutorial3scope1/di/UserComponent.java
package com.example.tutorial3scope1.di;


import com.example.tutorial3scope1.Activity1;
import com.example.tutorial3scope1.Activity4;
import com.example.tutorial3scope1.Activity2;
import com.example.tutorial3scope1.Activity3;

import dagger.Component;

@ActivityScope
@Component(modules = {UserModule.class})
public interface UserComponent {
    void inject(Activity1 activity1);

    void inject(Activity2 activity2);

    void inject(Activity3 activity3);

    void inject(Activity4 activity4);
}

File: Tutorial3-1Scope1/src/main/java/com/example/tutorial3scope1/di/UserComponent.kt
package com.example.tutorial3scope1.di

import com.example.tutorial3scope1.Activity1
import com.example.tutorial3scope1.Activity2
import com.example.tutorial3scope1.Activity3
import com.example.tutorial3scope1.Activity4
import dagger.Component

/**
 * @Provides method of module should have same scope with component, or both component and module
 * shouldn't have one
 */
@ActivityScope
@Component(modules = [UserModule::class])
interface UserComponent {

    fun inject(activity1: Activity1)
    fun inject(activity2: Activity2)
    fun inject(activity3: Activity3)
    fun inject(activity4: Activity4)
}